rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,371 @@
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withActivations(activations)
+                .build();
+
+            storageProvider.setOldFederation(oldFederation);
+            storageProvider.setNewFederation(newFederation);
+
+            Optional<Script> activeFederationRedeemScript = federationSupport.getActiveFederationRedeemScript();
+            assertTrue(activeFederationRedeemScript.isPresent());
+            assertThat(activeFederationRedeemScript.get(), is(expectedRedeemScript));
+        }
+
+        private Stream<Arguments> expectedRedeemScriptArgs() {
+            return Stream.of(
+                Arguments.of(oldFederation, null, genesisFederation.getRedeemScript()),
+                Arguments.of(null, newFederation, newFederation.getRedeemScript())
+            );
+        }
+
+        @ParameterizedTest
+        @Tag("getActiveFederationAddress")
+        @MethodSource("expectedAddressArgs")
+        void getActiveFederationAddress_returnsExpectedAddress(Federation oldFederation, Federation newFederation, Address expectedAddress) {
+            storageProvider.setOldFederation(oldFederation);
+            storageProvider.setNewFederation(newFederation);
+
+            Address activeFederationAddress = federationSupport.getActiveFederationAddress();
+            assertThat(activeFederationAddress, is(expectedAddress));
+        }
+
+        private Stream<Arguments> expectedAddressArgs() {
+            return Stream.of(
+                Arguments.of(oldFederation, null, genesisFederation.getAddress()),
+                Arguments.of(null, newFederation, newFederation.getAddress())
+            );
+        }
     }
 
-    private static Stream<Arguments> fedActivationAgeTestArgs() {
-        FederationConstants federationMainNetConstants = FederationMainNetConstants.getInstance();
-        FederationConstants federationTestNetConstants = FederationTestNetConstants.getInstance();
+    @Nested
+    @Tag("non null federations")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class ActiveFederationTestsWithNonNullFederations {
+        // new federation should be active if we are past the activation block number
+        // old federation should be active if we are before the activation block number
+        // activation block number is smaller for hop than for fingerroot
+
         ActivationConfig.ForBlock hopActivations = ActivationConfigsForTest.hop400().forBlock(0);
+        long blockNumberFederationActivationHop;
         ActivationConfig.ForBlock fingerrootActivations = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        long blockNumberFederationActivationFingerroot;
+        FederationStorageProvider storageProvider;
+
+        @BeforeAll
+        void setUp() {
+            long oldFederationCreationBlockNumber = 0;
```

Comment:
What do you think about setting a value greater than zero, like 200? Since this is not the genesis, it is not possible for it to be zero. Just to be more realistic.

Reply:
good one, thanks :)